### PR TITLE
Entity.resolve resolves defstruct correctly

### LINKED ISSUE #569: Entity.resolve fails on `defstruct`
given the following code:
```elixir
defmodule Struct do 
  defstruct [:foo]
```

`Entity.resolve` cannot resolve the call to defstruct on the second line, no matter where the cursor position is.

Expected behavior: `{:ok, {:call, Kernel, :defstruct, 2}}`. Actual behavior: `{:error, :not_found}`

### PR DESCRIPTION
**Pull request by scohen**
Fixes #569

### CODE REVIEW

File: apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex [hunk 1]
```diff
@@ -63,6 +63,10 @@ defmodule Lexical.RemoteControl.CodeIntelligence.Entity do
     resolve_alias(chars, node_range, analysis, position)
   end
 
+  defp resolve({:local_or_var, 'defstruct'}, node_range, _analysis, _position) do
```
Comment (zachallaun) [inline apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex:66]: Why does this need to be special-cased? Shouldn't this be resolved by looking at imports?

For instance, while ill-advised, one could shadow `defstruct` and use their own wrapper, e.g.

```elixir
defmodule MyModule do
  import Kernel, except: [defstruct: 1]
  import MyDefstruct
  
  # from MyDefstruct
  defstruct x: 1, y: 2
end
```

Comment (scohen) [reply apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex:66]: It's a little unclear to me how entity should handle this generically though. There's a local or var case already, but that seems to want to return a 0 arity call.

Comment (scohen) [reply apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex:66]: worse, i cannot find a way to reliably determine the arity of the function in question if it doesn't have parens.

Comment (scohen) [reply apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex]: ok, i took another stab at it.

**Pull request merged**